.github/workflows/pr-checks.yaml

```diff
@@ -0,0 +1,151 @@
+name: "Pull Request Checks"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+
+jobs:
+  pull-reqeust-checks:
+    needs:
+      - title-validation
+      - associated-issue
+    runs-on: ubuntu-22.04
+    if: always()
+    steps:
+      - if: contains(needs.*.result, 'failure')
+        run: echo "Failed due to ${{ contains(needs.*.result, 'failure') }}" && exit 1
```

Comment:
The reason we have the `ci` job in the other one is so we don't have to update admin settings to enable/disable branch protection; instead we have a single job listed as blocking CI. This workflow does not require such a marker job.

```suggestion
```

Reply:
Then I would need to set a required status check for each of those jobs I think.